src/ol_infrastructure/applications/xqueue/__main__.py

```diff
@@ -0,0 +1,228 @@
+from ol_infrastructure.components.aws.auto_scale_group import (
+    BlockDeviceMapping,
+    OLAutoScaleGroupConfig,
+    OLAutoScaling,
+    OLLaunchTemplateConfig,
+    TagSpecification,
+)
+from ol_infrastructure.lib.aws.ec2_helper import InstanceTypes, default_egress_args
+from ol_infrastructure.lib.consul import get_consul_provider
+from ol_infrastructure.lib.ol_types import AWSBase
+from ol_infrastructure.lib.pulumi_helper import parse_stack
+from ol_infrastructure.lib.vault import setup_vault_provider
+
+stack_info = parse_stack()
+xqueue_config = Config("xqueue")
+if Config("vault").get("address"):
+    setup_vault_provider()
+
+network_stack = StackReference(f"infrastructure.aws.network.{stack_info.name}")
+policy_stack = StackReference("infrastructure.aws.policies")
+dns_stack = StackReference("infrastructure.aws.dns")
+consul_stack = StackReference(
+    f"infrastructure.consul.{stack_info.env_prefix}.{stack_info.name}"
+)
+
+vault_stack = StackReference(f"infrastructure.vault.operations.{stack_info.name}")
+edxapp = StackReference(
+    f"applications.edxapp.{stack_info.env_prefix}.{stack_info.name}"
+)
+
+env_name = f"{stack_info.env_prefix}-{stack_info.env_suffix}"
+target_vpc_name = xqueue_config.get("target_vpc")
+target_vpc = network_stack.require_output(target_vpc_name)
+
+aws_account = get_caller_identity()
+vpc_id = target_vpc["id"]
+xqueue_server_ami = ec2.get_ami(
+    filters=[
+        ec2.GetAmiFilterArgs(name="name", values=["open-edx-xqueue-server-*"]),
+        ec2.GetAmiFilterArgs(name="virtualization-type", values=["hvm"]),
+        ec2.GetAmiFilterArgs(name="root-device-type", values=["ebs"]),
+    ],
+    most_recent=True,
+    owners=[aws_account.account_id],
+)
+aws_config = AWSBase(
+    tags={
+        "OU": xqueue_config.require("business_unit"),
+        "Environment": env_name,
+        "Application": "open-edx-xqueue",
+        "Owner": "platform-engineering",
+    }
+)
+xqueue_server_tag = f"open-edx-xqueue-server-{env_name}"
+consul_security_groups = consul_stack.require_output("security_groups")
+consul_provider = get_consul_provider(stack_info)
+
+# IAM and instance profile
+xqueue_server_instance_role = iam.Role(
+    f"xqueue-server-instance-role-{env_name}",
+    assume_role_policy=json.dumps(
+        {
+            "Version": "2012-10-17",
+            "Statement": {
+                "Effect": "Allow",
+                "Action": "sts:AssumeRole",
+                "Principal": {"Service": "ec2.amazonaws.com"},
+            },
+        }
+    ),
+    path=f"/ol-applications/open-edx-xqueue/{stack_info.env_prefix}/{stack_info.env_suffix}/",
+    tags=aws_config.tags,
+)
+# Vault policy definition
+xqueue_vault_policy = vault.Policy(
+    "open-edx-xqueue-server-vault-policy",
+    name=f"xqueue-{stack_info.env_prefix}",
+    policy=Path(__file__)
+    .parent.joinpath("xqueue_policy.hcl")
+    .read_text()
+    .replace("DEPLOYMENT", f"{stack_info.env_prefix}"),
+)
+print(
```

Comment:
Please remove print calls before merging

Reply:
Not good to have prints for bug processeS?